src/factories/calorimetry/CalorimeterClusterRecoCoG_factoryT.h

```diff
@@ -48,15 +60,25 @@ class CalorimeterClusterRecoCoG_factoryT :
         // SpdlogMixin logger initialization, sets m_log
         InitLogger(app, GetPrefix(), "info");
 
-        // Algorithm configuration
-        auto cfg = GetDefaultConfig();
-
-        app->SetDefaultParameter(param_prefix + ":samplingFraction", cfg.sampFrac);
-        app->SetDefaultParameter(param_prefix + ":logWeightBase", cfg.logWeightBase);
-        app->SetDefaultParameter(param_prefix + ":energyWeight", cfg.energyWeight);
-        app->SetDefaultParameter(param_prefix + ":enableEtaBounds", cfg.enableEtaBounds);
+        // Initialize properties
+        for (const auto& [key, prop] : m_algo.getProperties()) {
+          std::visit(
+            [app, &m_algo = m_algo, &m_log = m_log,
+             param_prefix, key = key](auto&& val) {
```

Comment:
I might be wrong, but this could be just
```suggestion
            [app, &m_algo, &m_log, param_prefix, key](auto& val) {
```

Reply:
"Class members cannot be captured explicitly by a capture without initializer (as mentioned above, only variables are permitted in the capture list)" (somewhere in the long page https://en.cppreference.com/w/cpp/language/lambda)

We could capture `this`, but I wanted to avoid that (more scope than needed). It would look like this:
```suggestion
            [app, this, param_prefix, key = key](auto&& val) {
```

Reply:
`key` is not a variable either; it's a structured binding.

Reply:
I looked up the structured bindings in captures rules, and it turns out that's allowed in C++20 now. About 2/3 down the page at https://en.cppreference.com/w/cpp/language/structured_binding.